### PR TITLE
Calculate more literal values

### PR DESCRIPTION
**Pull request by orklah**
This is an attempt to fix https://github.com/vimeo/psalm/issues/5021

I tweaked a condition slightly and it immediately gave me the value I expected. It seem way to easy to be right but I guess CI will tell me!

### CODE REVIEW

**Comment by orklah** [comment]
Looks like everything pass, this is cool !

File: tests/BinaryOperationTest.php [hunk 1]
```diff
@@ -196,6 +196,15 @@ function foo(string $s) : int {
                     '$f' => 'string',
                 ],
             ],
+            'ComplexLiteralBitwise' => [
+                '<?php
+                    /**
+                     * @return 7
+                     */
+                    function scope(){
+                        return 1 | 2 | 4;
```
Comment (weirdan) [inline tests/BinaryOperationTest.php:205]: I would suggest to also add a test with `0`, bitwise and (`&`), and brackets.

File: src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php [hunk 2]
```diff
@@ -295,13 +295,14 @@ private static function analyzeNonDivOperands(
             && $right_type_part instanceof TLiteralInt
             && ($left instanceof PhpParser\Node\Scalar
                 || $left instanceof PhpParser\Node\Expr\ConstFetch
-                || $left instanceof PhpParser\Node\Expr\ClassConstFetch)
+                || $left instanceof PhpParser\Node\Expr\ClassConstFetch
+                || $left instanceof PhpParser\Node\Expr\BinaryOp)
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php:299]: Shouldn't this be any expression really? E.g. `(true ? 1 : 2) | 22` would not have `$left` as binary op, but it still can be calculated, I suppose.

**Review by weirdan — CHANGES_REQUESTED** [review]

File: tests/BinaryOperationTest.php (hunk 1 above)
Comment (orklah) [reply tests/BinaryOperationTest.php]: Done

File: src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php (hunk 2 above)
Comment (orklah) [reply src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php:299]: Well, we saw a good example of 1 | 2 | 4 in the form of flags and we need Psalm to understand it to be coherent. I'm not sure if it's worth adding complexity and risking edge cases for  `(true ? 1 : 2) | 22`. I'd be open to consider this if we find a real case where it change the outcome of psalm's analysis

Comment (weirdan) [reply src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php:299]: I can imagine things like ` & ~$this->flagMask `, but you're right, we can fix it when we see it.

**Review by weirdan — APPROVED** [review]

**Comment by muglug** [comment]
Thanks!

**Pull request merged**